nncf/experimental/openvino_native/quantization/algorithms/min_max/openvino_backend.py

```diff
@@ -172,32 +175,59 @@ def _get_num_samples(num_samples, target_point: OVTargetPoint):
         return num_samples
 
     @staticmethod
-    def minmax_statistic_collector(nncf_graph: NNCFGraph,
-                                   target_point: OVTargetPoint,
-                                   quantizer_config: QuantizerConfig,
-                                   inplace: bool,
-                                   num_samples: int = None,
-                                   ) -> TensorCollector:
+    def get_statistic_collector(range_estimator_params: RangeEstimatorParameters,
+                                nncf_graph: NNCFGraph,
+                                target_point: OVTargetPoint,
+                                quantizer_config: QuantizerConfig,
+                                inplace: bool,
+                                num_samples: int = None) -> TensorCollector:
         reduction_shape, use_abs_max =\
             OVMinMaxAlgoBackend._get_reduction_shape_and_use_abs_max(nncf_graph, target_point,
                                                                      quantizer_config)
         _num_samples = OVMinMaxAlgoBackend._get_num_samples(num_samples, target_point)
-        return get_min_max_stat_collector(_num_samples, reduction_shape, use_abs_max, inplace)
 
-    @staticmethod
-    def mean_minmax_statistic_collector(nncf_graph: NNCFGraph,
-                                        target_point: OVTargetPoint,
-                                        quantizer_config: QuantizerConfig,
-                                        use_per_sample_stats: bool,
-                                        inplace: bool,
-                                        num_samples: int = None,
-                                        ) -> TensorCollector:
-        reduction_shape, use_abs_max = \
-            OVMinMaxAlgoBackend._get_reduction_shape_and_use_abs_max(nncf_graph, target_point,
-                                                                     quantizer_config)
-        _num_samples = OVMinMaxAlgoBackend._get_num_samples(num_samples, target_point)
-        return get_mean_min_max_stat_collector(_num_samples, reduction_shape,
-                                               use_abs_max, use_per_sample_stats, inplace)
+        try:
+            base_kwargs = {
+                'reduction_shape': reduction_shape,
+                'inplace': inplace
+            }
+            reducers = []
+            for i, params in enumerate([range_estimator_params.min, range_estimator_params.max]):
+                kwargs = base_kwargs.copy()
+                if params.statistics_type in [StatisticsType.QUANTILE,
+                                              StatisticsType.ABS_QUANTILE]:
+                    kwargs.update({'quantile': [i + (1 - 2 * i) * params.quantile_outlier_prob]})
+                # TODO(dlyakhov): merge two quantile aggregators in one
+                statistic_type = params.statistics_type
+                if use_abs_max and statistic_type == StatisticsType.MAX:
```

Comment:
Could you please explain why you change `statistic_type` by `use_abs_max` flag?

Reply:
The logic in MinMax is following: common algorithm asks for a TensorCollector (MinMax or MeanMinMax), and then the backend decides should `abs_max` be used or not. Is this logic changed?

Reply:
Ticket on refactoring: 109437